src/Fmt_ast.ml

```diff
@@ -631,20 +631,18 @@ let doc_atrs atrs =
   (doc, List.rev rev_atrs)
 
 let fmt_docstring c ?pro ?epi doc =
-  opt doc (fun (({txt; loc} as doc), floating) ->
+  opt doc (fun ({txt; loc}, floating) ->
       let epi =
         match epi with
         | Some _ -> epi
         | None when floating -> Some (fmt "@,")
         | None -> None
       in
-      fmt_if_k
-        (not (Cmts.doc_is_dup c.cmts doc))
```

Comment:
@gpetiot this stale code is what was causing the lost docstrings in #303

Reply:
I wonder if we should fix it instead . I expect this line to be useful together with [--no-check-comments]

Reply:
Perhaps, I'm not sure. It depends on the intended use of `--no-check-comments` I think. To me, it is basically just a debug option to see what the output is, but it never to be actually used. For only that, I'd rather have simpler code with less global state. But maybe you have other uses of `--no-check-comments`?

Reply:
Well, I have been using that option the format the whole compiler code base (https://github.com/hhugo/ocaml/tree/auto) as an experiment/demo.
Thinking more about it, I think this dedup, if any, should be done using an ast_mapper.

Reply:
Yes, as an ast_mapper it would be more predictable wrt which of the duplicates is actually printed, rather than relying on whichever one happened to be hit first by the evaluation of what ought to be a formatting stream * state monad.